services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentMigratorITest.kt

```diff
@@ -0,0 +1,146 @@
+    }
+
+    @Nested
+    @Transactional
+    inner class FindAttachmentWithDatabaseContent {
+        @Test
+        fun `Should return null if no attachments in content table`() {
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()
+
+            assertThat(attachmentWithContent).isNull()
+        }
+
+        @Test
+        fun `Should return un-migrated content when there are any`() {
+            val attachment = attachmentFactory.save().withDbContent().value
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            val content = contentRepository.findByIdOrNull(attachment.id!!)!!
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
+
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()
+
+            assertThat(attachmentWithContent).isNotNull().all {
+                prop(ApplicationAttachmentWithContent::id).isEqualTo(attachment.id!!)
+                prop(ApplicationAttachmentWithContent::applicationId)
+                    .isEqualTo(attachment.applicationId)
+                prop(ApplicationAttachmentWithContent::content).all {
+                    prop(AttachmentContent::fileName).isEqualTo(attachment.fileName)
+                    prop(AttachmentContent::contentType).isEqualTo(attachment.contentType)
+                    prop(AttachmentContent::bytes).isEqualTo(content.content)
+                }
+            }
+        }
+    }
+
+    @Nested
+    inner class Migrate {
+        @Test
+        fun `Should migrate content to blob`() {
+            attachmentFactory.save().withDbContent().value
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()!!
+
+            val blobPath = attachmentMigrator.migrate(attachmentWithContent)
+
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET).first()).all {
+                prop(TestFile::path).isEqualTo(blobPath)
+            }
+        }
+    }
+
+    @Nested
+    @Transactional
+    inner class SetBlobPathAndCleanup {
+        @Test
+        fun `Should throw AttachmentNotFoundException if attachment is not in db`() {
+            val attachment = attachmentFactory.save().withDbContent().value
+            val blobPath = generateBlobPath(attachment.applicationId)
+            attachmentRepository.delete(attachment)
+            assertThat(attachmentRepository.findAll()).isEmpty()
```

Comment:
Is it necessary to save an attachment and then delete it? It's the same as if it never existed. Except that `attachmentFactory.save()` saves an application, but `setBlobPathAndCleanup` doesn't use the application for anything.
```suggestion
            val attachmentId = UUID.fromString("1695d310-0602-4822-9770-ff76847a4479")
            val blobPath = "15/$attachmentId"
```

Reply:
Fixed.